### PR TITLE
Added support for autodetecting the system monitor color profile also on Linux

### PR DESCRIPTION
**Pull request by agriggio**
Code borrowed from Geeqie.

Works fine here, but please test in other environments, particularly on Windows since I don't have that (to make sure I did not break anything there)

### CODE REVIEW

**Comment by Beep6581** [comment]
@agriggio I just wanted to make sure that the code is not from sourceforge but from http://geeqie.org/cgi-bin/gitweb.cgi?p=geeqie.git Sourceforge is obsolete but they can't remove the project for some reason, the git link above should be current. Their official website is http://geeqie.org/ (I updated it for them). I can confirm that monitor profile detection works in Geeqie-git in Linux - haven't tried this RT branch yet.

**Comment by agriggio** [comment]
I've actually taken this from the latest release, not the git master. I'll
check if anything has changed and update the patch if needed. it works here
fwiw

**Comment by agriggio** [comment]
@Beep6581 I confirm that the current git master uses the exact same code as the latest release for detecting the monitor color profile

**Comment by gaaned92** [comment]
@agriggio no modification  for profile selection compared to dev branch on W10.
But I think there is a bug.
if  in the "option file, color management part" you have for instance
```
ICCDirectory=C:\\Windows\\System32\\spool\\drivers\\color
MonitorProfile=Rec2020
AutoMonitorProfile=true
```
instead of selecting the default system profile, the indicated profile is selected.

I observe that the selection at the bottom of editor panel is reset to the option file selection at RT start. It is logical but should be documented in rawpedia.

**Comment by agriggio** [comment]
@gaaned92 thanks for testing. the change in behaviour is not intentional,
I'll see what is going on.

**Comment by gaaned92** [comment]
@agriggio I  tested with dev branch and behaviour is the same, so it is certainly an old bug.

File: rtgui/editorpanel.cc [hunk 1]
```diff
@@ -105,7 +105,7 @@ bool find_default_monitor_profile(GdkWindow *rootwin, Glib::ustring &defprof, Gl
     // * Copyright (C) 2006 John Ellis
     // * Copyright (C) 2008 - 2016 The Geeqie Team
     // 
-    guchar *prof;
+    guchar *prof = NULL;
```
Comment (Floessie) [inline rtgui/editorpanel.cc]: @agriggio Please use `nullptr` instead. I converted all places with `NULL` prior to the 5.0 release and it should stay consistent from now on.

**Comment by agriggio** [comment]
> so it is certainly an old bug.

@gaaned92 can you open a separate issue then? Thanks!

File: rtgui/editorpanel.cc (hunk 1 above)
Comment (agriggio) [reply rtgui/editorpanel.cc]: @Floessie sure :+1:

**Comment by agriggio** [comment]
@Floessie I replaced `NULL` with `nullptr`. But FYI, I still see several uses of `NULL` in the code (unrelated to this patch). Should an issue be opened to fix them as well?

**Comment by Floessie** [comment]
@agriggio Thanks for the fix. I'll care about the other places (and `0`) as soon as I'm back from my holydays.

**Comment by agriggio** [comment]
any objections to merging this? According to @gaaned92 the behaviour on windows is unchanged, and on Linux it seems to work fine (tested on two different machines with two different desktop environments)

**Comment by Beep6581** [comment]
Auto-detection works, but if I set the monitor default color profile to Foo, then enable "Use operating system's main monitor color profile" and open an image, the monitor profile selected by default under the preview is still Foo. Even if I change it to "System default (from GDK)" and restart RT, it still defaults to Foo. Since the Preferences options is called "monitor **default** color profile", "System default (from GDK)" should be on by default regardless of what is selected in the disabled combo.

**Comment by agriggio** [comment]
@Beep6581 that's the same behaviour described by @gaaned92 I think. It seems this is a bug that is already in dev. I can try fixing that, but I propose we merge first. What do you think?

**Comment by Beep6581** [comment]
@agriggio :+1:

**Comment by Beep6581** [comment]
I opened a new issue #3821

**Pull request merged**